sdk/core/core-rest-pipeline/src/browserHttpClient.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HttpClient, PipelineRequest, PipelineResponse } from "./interfaces";
+import { createFetchHttpClient } from "./fetchHttpClient";
+import { createXhrHttpClient } from "./xhrHttpClient";
+
+/**
+ * Create an HTTP Client that uses either XHR or fetch.
+ * Once a single request enables streaming, fetch will be used.
+ */
+export function createBrowserHttpClient(): HttpClient {
+  let client: HttpClient;
```

Comment:
nit: does not change anything since you already handled the `undefined` case, but morally, it is initially set to `undefined`:
```suggestion
  let client: HttpClient | undefined;
```

Reply:
Hm won't this allow `undefined` to be re-assigned to the variable? I think not declaring `| undefined` is fine as long as TS can analyze that it's assigned before use.

Reply:
You're right, I did not notice that the compiler is that smart!